### PR TITLE
Add smoke tests

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Exclude more files from Docker build context
- Add scripts to run tests
- Add smoke tests to CI
- Move `input.ex` files to `lib` directory for test data

I've tried to keep the existing structure as much the same as possible.

### CODE REVIEW

**Comment by ErikSchierboom** [comment]
@angelikatyborska Would you mind taking a look?

**Comment by angelikatyborska** [comment]
I'm pretty busy in the next 7 days, maybe @jiegillet can review?

File: bin/run-tests.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    test_dir_path=$(realpath "${test_dir}")
+
+    bin/run.sh "${test_dir_name}" "${test_dir_path}" "${test_dir_path}"
+
+    file="representation.txt"
+    expected_file="expected_representation.ex"
+    echo "${test_dir_name}: comparing ${file} to ${expected_file}"
+
+    if ! diff "${test_dir_path}/${file}" "${test_dir_path}/${expected_file}"; then
+        exit_code=1
```
Comment (jiegillet) [inline bin/run-tests.sh:37]: Can we echo a something about the file being different here? Easier to make sense out of a failure

File: .github/workflows/smoke_test.yml [hunk 2]
```diff
@@ -21,3 +21,27 @@ jobs:
       - name: Run Tests
         run: |
           mix test
+
+  build_and_test_on_docker:
+    runs-on: ubuntu-22.04
```
Comment (jiegillet) [inline .github/workflows/smoke_test.yml:26]: The `build_and_test` job above runs on `ubuntu-20.04`. While you're here, would you mind bumping it?

File: bin/run-tests.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env sh
```
Comment (jiegillet) [inline bin/run-tests.sh:1]: Would it be cool to rename `bin/run-tests-in-docker.sh` and `bin/run-tests.sh` => `bin/run-smoke-tests-in-docker.sh` and `bin/run-smoke-tests.sh`? 
The Elixir tests and the smoke test happen to pretty much do the same thing, but I would prefer keeping the distinction.

**Review by jiegillet — COMMENTED** [review]
Looks good, thank you for doing this, and sorry about the `lib` thing, I bet it took a while to figure out!

I left some minor requests.

File: bin/run-tests.sh (hunk 1 above)
Comment (ErikSchierboom) [reply bin/run-tests.sh:37]: Well, the line above has: 

> echo "${test_dir_name}: comparing ${file} to ${expected_file}"

Wouldn't that be enough?

File: bin/run-tests.sh (hunk 3 above)
Comment (ErikSchierboom) [reply bin/run-tests.sh:1]: Well, we could, but I would personally like to keep its current name for consistency with other repos.

Comment (jiegillet) [reply bin/run-tests.sh:1]: Right, I realized later that this was the norm. Ok that's fine :)

File: bin/run-tests.sh (hunk 1 above)
Comment (jiegillet) [reply bin/run-tests.sh:37]: No, because it won't exit here right? It only sets `exit_code=1`, so if something fails we won't know where that happened

Comment (ErikSchierboom) [reply bin/run-tests.sh:37]: This is an example of what the output looks like when there is a mismatch:

```
user_defined_structs: comparing representation.txt to expected_representation.ex
2c2
<   defstruct [:wheels, :doors, :color]
---
>   struct [:wheels, :doors, :color]
```

That's not enough you think?

Comment (jiegillet) [reply bin/run-tests.sh:37]: Oh, sorry I didn't realize we would get the output from the `diff`. Yes, thank you that's plenty.

**Comment by ErikSchierboom** [comment]
Thanks all!

**Pull request merged**